DQMOffline/L1Trigger/interface/HistDefinition.h

```diff
@@ -0,0 +1,32 @@
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+namespace dqmoffline {
+namespace l1t {
+
+class HistDefinition;
+typedef std::map<std::string, HistDefinition> HistDefinitions;
```

Comment:
hi @kreczko - as the possible list of strings are in any case hardwired, please change std::string to an enum type (that needs then a once per job conversion from string->enum when interpreting the PSet.

Reply:
Hi @davidlange6 thanks for the feedback.

What kind of implementation would you suggest? 
The examples I can think of are
```c++
// in each header file (so 3x right now)
// add per module plotConfig
enum PlotConfig {
    nVertex // add more if needed
  };

// function to do the mapping
static std::vector<HistDefinition> readHistDefinitions(const edm::ParameterSet &ps);

// change
dqmoffline::l1t::HistDefinitions histDefinitions_;
// to
std::vector<HistDefinition> histDefinitions_;
```
or
```c++
// change
typedef std::map<std::string, HistDefinition> HistDefinitions;
// to
typedef std::vector<HistDefinition> HistDefinitions;
// change
HistDefinitions dqmoffline::l1t::readHistDefinitions(const edm::ParameterSet &ps);
// to
HistDefinitions dqmoffline::l1t::readHistDefinitions(const edm::ParameterSet &ps, std::map<std::string, unsigned int>);

// and add to each header

enum PlotConfig {
    nVertex // add more if needed
  };

// pseudocode
static const std::map<std::string, unsigned int> PlotConfigNames;
...
const std::map<std::string, unsigned int> MyModule::PlotConfigNames =  {
      { "nVertex", PlotConfig::nVertex }
};
```
or maybe a solution using `constexpr`?

Reply:
hi @kreczko - Right, something like those - the second keeps the functionality of being able to make plots of a subset of those possible.